### PR TITLE
Newsletters: add 66 (2019-10-02)

### PR DESCRIPTION
**Pull request by harding**

### CODE REVIEW

File: _posts/en/newsletters/2019-10-02-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    been disclosed, a [PR][BOLTS #676] has been opened to update the
+    specification to note that this check is needed.
+
+- **Optech taproot workshop:** last week, Optech held workshops in
+  both San Francisco and New York City teaching developers about the
+  schnorr signature scheme and other parts of the proposed
+  [taproot][bip-taproot] soft fork.  We thank Bryan Bishop for typing an
+  excellent [transcript][workshop transcript] of the NYC workshop.  We
+  are preparing videos and other educational material for release via a
+  blog post in the near future.
```
Comment (moneyball) [inline _posts/en/newsletters/2019-10-02-newsletter.md]: Do we want to link to https://github.com/bitcoinops/taproot-workshop or save this for the upcoming blog post?

File: _posts/en/newsletters/2019-10-02-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+title: 'Bitcoin Optech Newsletter #66'
+permalink: /en/newsletters/2019/10/02/
+name: 2019-10-02-newsletter
+slug: 2019-10-02-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter notes a proposed BIP that would allow nodes to
+implement the erlay transaction relay protocol, announces full
```
Comment (jonatack) [inline _posts/en/newsletters/2019-10-02-newsletter.md]: nit regarding "erlay" here, and the two other cases in this newsletter: the authors call it "Erlay" in their paper last June (https://arxiv.org/pdf/1905.10518.pdf) and git grepping this repo returns only "Erlay" and no other instances of "erlay"

File: _posts/en/newsletters/2019-10-02-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+disclosure of vulnerabilities that affected earlier versions of LN
+implementations, and links to a transcript of Optech's recent schnorr
+and taproot workshop.  Also included is the description of several
```
Comment (jonatack) [inline _posts/en/newsletters/2019-10-02-newsletter.md]: s/workshop/workshops/?

File: _posts/en/newsletters/2019-10-02-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  the BIP authors either privately or on the mailing list.
+
+- **Full disclosure of fixed vulnerabilities affecting multiple LN
+  implementations:** several weeks ago, the developers of C-Lightning,
+  Eclair, and LND announced the previous discovery of an undisclosed
+  issue in each of their implementations, which they had each fixed in a
+  recent release.  At that time, they strongly encouraged their users to
+  upgrade (a message which Optech [relayed][optech ln warning]) and
+  promised a full disclosure in the future, which they have now done
+  with a [email][russell disclosure] by vulnerability discoverer and
```
Comment (jonatack) [inline _posts/en/newsletters/2019-10-02-newsletter.md]: s/a email/an email/

File: _posts/en/newsletters/2019-10-02-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+  C-Lightning developer Rusty Russell and a [blog post][lnd stay safe]
+  by LND developers Olaoluwa Osuntokun and Conner Fromknecht.
+
+    Briefly, the issue appears to have been that implementations did not
```
Comment (jonatack) [inline _posts/en/newsletters/2019-10-02-newsletter.md]: missing "the" between "that" and "implementations"

File: _posts/en/newsletters/2019-10-02-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    confirm that channel open transactions paid the correct script,
+    amount, or both.  Because of this, the implementations would accept
+    payments within the channel which they would later be unable to get
+    confirmed onchain, allow them to be defrauded.  As of this writing,
```
Comment (jonatack) [inline _posts/en/newsletters/2019-10-02-newsletter.md]: s/allow/allowing/?

File: _posts/en/newsletters/2019-10-02-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+{% comment %}<!-- include references.md below the fold but above any Jekyll/Liquid variables-->{% endcomment %}
+{% include references.md %}
+
+## Action items
+
+*None this week.*
+
+## News
+
+- **Draft BIP for enabling erlay compatibility:** a draft [BIP for
```
Comment (jonatack) [inline _posts/en/newsletters/2019-10-02-newsletter.md]: Perhaps mention the BIP's title, "Transaction announcements reconciliation"

File: _posts/en/newsletters/2019-10-02-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+  blog post in the near future.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[LND][lnd repo], [C-Lightning][c-lightning repo], [Eclair][eclair repo],
+[libsecp256k1][libsecp256k1 repo], [Bitcoin Improvement Proposals
+(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #15558][] changes the number of DNS seeds Bitcoin Core
```
Comment (jonatack) [inline _posts/en/newsletters/2019-10-02-newsletter.md]: Note: interesting how much discussion this PR got, and ajtowns' post-merge review suspecting a bug. Might be a good review club PR.

File: _posts/en/newsletters/2019-10-02-newsletter.md (hunk 6 above)
Comment (jonatack) [reply _posts/en/newsletters/2019-10-02-newsletter.md]: or s/allow/and allow/

File: _posts/en/newsletters/2019-10-02-newsletter.md (hunk 1 above)
Comment (jnewbery) [reply _posts/en/newsletters/2019-10-02-newsletter.md]: I think save it. Let's release/announce all the material at the same time.

File: _posts/en/newsletters/2019-10-02-newsletter.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+    confirm that channel open transactions paid the correct script,
+    amount, or both.  Because of this, the implementations would accept
+    payments within the channel which they would later be unable to get
+    confirmed onchain, allow them to be defrauded.  As of this writing,
+    Optech is not aware of any reports that this issue was exploited
+    prior to the warning last month.  Now that the issue has
+    been disclosed, a [PR][BOLTS #676] has been opened to update the
+    specification to note that this check is needed.
+
+- **Optech taproot workshop:** last week, Optech held workshops in
```
Comment (bitschmidty) [inline _posts/en/newsletters/2019-10-02-newsletter.md]: Half the workshop was on schnorr so perhaps its more accurate to title as 'Optech schnorr and taproot workshop'

File: _posts/en/newsletters/2019-10-02-newsletter.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+  about other possible peers, with the node saving this information to
+  disk for use later (including after restarts).  Ideally, nodes only
+  ever query DNS seeds once---when they're first started.  Practically,
+  though, they may also query on subsequent startups if none of the
+  saved peers they select responds quickly enough.
+
+    This merge causes Bitcoin Core to only query three DNS seeds at a
+    time rather than all of them.  Three seeds should be enough
+    diversity in sources to ensure the node connects to at least one
+    honest peer (a requirement for Bitcoin security) but it's few enough
```
Comment (adamjonas) [inline _posts/en/newsletters/2019-10-02-newsletter.md]: security) but -> security), but

File: _posts/en/newsletters/2019-10-02-newsletter.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+- **Draft BIP for enabling erlay compatibility:** a draft [BIP for
+  transaction relay through set reconciliation][bip-reconcil] has been
+  [posted][reconcil post] to the Bitcoin-Dev mailing list by [erlay][]
+  co-author Gleb Naumenko.  Currently, Bitcoin nodes send each of their
+  peers the txids for all newly-seen transactions, resulting in each
+  node receiving many duplicate txid announcements.  This can be quite
+  bandwidth inefficient for Bitcoin's tens of thousands of nodes and
+  several hundred thousand transactions a day.  As [described
+  previously][erlay paper], minisketch-based set reconciliation allows
+  nodes to instead send a *sketch* of a set of txids that can be combined
```
Comment (adamjonas) [inline _posts/en/newsletters/2019-10-02-newsletter.md]: nit: to instead send -> to send (split infinitive and instead doesn't add extra meaning.)

**Comment by harding** [comment]
Updated with BTSE report and to address feedback (thanks!).

File: _posts/en/2019-10-01-btse-exchange-operation.md [hunk 12]
```diff
@@ -0,0 +1,20 @@
+---
+title: 'Field Report: Exchange operation using Bitcoin technologies at BTSE'
+permalink: /en/btse-exchange-operation/
+name: 2019-10-??-btse-exchange-operation
```
Comment (bitschmidty) [inline _posts/en/2019-10-01-btse-exchange-operation.md]: TO FIX

**Comment by harding** [comment]
Made the changes suggested in https://github.com/bitcoinops/bitcoinops.github.io/pull/230#pullrequestreview-295923772 and squashed.

**Comment by bitschmidty** [comment]
I noticed that 2 of the links in the field report were absolute so I corrected those to relative (which required adding one of them to the references.md)

**Comment by bitschmidty** [comment]
Last week we had the Optech workshop
Transcribed in NY by Bryan Bishop

BTSE reports on using bitcoin tech
Do it yourself so you dont get wrecked

Erlay suggests efficiency/bandwidth mix
LN vulnerabilities they have a fix

**Pull request merged**